datahub/search/management/commands/sync_es.py

```
@@ -0,0 +1,179 @@
+                   fn=lambda col: list(map(_id_type_dict, col.all()))),
+    FieldsColumnFn(fields=('contacts',),
+                   fn=lambda col: list(map(_contact_dict, col.all()))),
+    FieldsColumnFn(fields=('id',),
+                   fn=lambda col: str(col)),
+    FieldsColumnFn(fields=('export_to_countries', 'future_interest_countries',),
+                   fn=lambda col: list(map(_id_name_dict, col.all()))),
+)
+
+# there is no typo in 'servicedeliverys' :(
+_ignored_fields = (
+    'subsidiaries', 'servicedeliverys', 'investment_projects',
+    'investor_investment_projects', 'intermediate_investment_projects',
+    'investee_projects', 'recipient_investment_projects', 'teams',
+)
+
+
+def get_dataset():
+    """Returns dataset that will be synchronised with Elasticsearch."""
+    return (
+        DataSet(Company.objects.all(), ESCompany),
+        DataSet(Contact.objects.all(), ESContact),
+    )
+
+
+def _id_name_dict(obj):
+    """Creates dictionary with selected field from supplied object."""
+    return {
+        'id': str(obj.id),
+        'name': obj.name,
+    }
+
+
+def _id_type_dict(obj):
+    """Creates dictionary with selected field from supplied object."""
+    return {
+        'id': str(obj.id),
+        'type': obj.type
+    }
+
+
+def _contact_dict(obj):
+    """Creates dictionary with selected field from supplied object."""
+    return {
+        'id': str(obj.id),
+        'first_name': obj.first_name,
+        'last_name': obj.last_name,
+    }
+
+
+def _company_dict(obj):
+    return {
+        'id': str(obj.id),
+        'company_number': obj.company_number,
+    }
+
+
+def _model_to_dict(model, fields_column_fn=fields_column_fn):
+    """Converts model instance to a dictionary suitable for ElasticSearch."""
+    result = {}
+    for fcf in fields_column_fn:
+        result.update({field: fcf.fn(getattr(model, field))
+                       for field in fcf.fields if hasattr(model, field) and getattr(model, field)})
```

Isn't `if getattr(model, field, None)` the same as `if hasattr(model, field) and getattr(model, field)`?

Anyway, can't we define a `_fields_column_fn` per model? That should avoid the need for hasattr() or setattr() + default which might mask bugs.